### PR TITLE
Path payment test fixes

### PR DESCRIPTION
**Pull request by rokopt**
# Description

Split "offers in a loop" path payment tests into separate TEST_CASES
so that they can use Application objects without USE_CONFIG_FOR_GENESIS.

The path payment and strict path payment "use all offers in a loop"
tests perform modifications (an account merge and a ledger close)
in a for_versions() loop when and only when the protocol version is
13 or greater.  In master currently, it is 13, so the modifications are
performed only in version 13, which, because it is the current version,
is also the maximum protocol version of any bucket.

When the version is bumped to 14, however, there will be version-14
buckets from previous tests when the "use all offers in a loop" tests
try to perform modifications while running protocol 13.  This will
cause the bucket merge code to crash, as it will find existing buckets
of version 14 but it will be running with a maximum supported
protocol of 13 (during that iteration of for_versions()).

Other tests have faced this problem, and there is a test-only Config
option which works around it: disabling USE_CONFIG_FOR_GENESIS,
which those other tests do.

However, many of the path payment and strict path payment tests
depend upon USE_CONFIG_FOR_GENESIS, so we can't just turn it
off throughout all the path payment tests without changing many
other tests.  Therefore, this pull request fixes the problem by splitting
the "use all offers in a loop" tests, which are currently SECTION()s,
into their own TEST_CASE()s, with their own Application objects,
which use Config objects with USE_CONFIG_FOR_GENESIS disabled.

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v5.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [x] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

(Regarding performance, only very short-running tests are changed.)

### CODE REVIEW

File: src/transactions/test/PathPaymentStrictSendTests.cpp [hunk 1]
```diff
@@ -2434,3 +2332,132 @@ TEST_CASE("pathpayment strict send", "[tx][pathpayment]")
...
+                                 .offers;
+                });
+            auto expected = std::vector<ClaimOfferAtom>{
+                o1.exchanged(640, 1280), o2.exchanged(320, 640),
+                o3.exchanged(160, 320),  o4.exchanged(80, 160),
+                o1.exchanged(40, 80),    o2.exchanged(20, 40),
+                o3.exchanged(10, 20)};
+            checkClaimedOffers(actual, expected, 1280, 10);
+            // clang-format off
+                market.requireBalances(
```
Comment (jonjove) [inline src/transactions/test/PathPaymentStrictSendTests.cpp:2438]: Nitpick: gained an extra indent here

File: src/transactions/test/PathPaymentStrictSendTests.cpp [hunk 2]
```diff
@@ -2434,3 +2332,132 @@ TEST_CASE("pathpayment strict send", "[tx][pathpayment]")
...
+        for_versions_from(12, *app, [&] {
+            auto market = TestMarket{*app};
+            auto source = root.create("source", minBalance4);
+            auto destination = root.create("destination", minBalance1);
+            auto mm12 = root.create("mm12", minBalance3);
+            auto mm23 = root.create("mm23", minBalance3);
+            auto mm34 = root.create("mm34", minBalance3);
+            auto mm41 = root.create("mm41", minBalance3);
+            auto xlm = makeNativeAsset();
+            auto idr = makeAsset(gateway, "IDR");
```
Comment (jonjove) [inline src/transactions/test/PathPaymentStrictSendTests.cpp:2368]: Nitpick: `idr` and `usd` are unused.

File: src/transactions/test/PathPaymentTests.cpp [hunk 3]
```diff
@@ -4442,3 +4339,147 @@ TEST_CASE("pathpayment", "[tx][pathpayment]")
...
+    auto const morePayment = paymentAmount / 2;
+
+    // sets up gateway account
+    auto const gatewayPayment = minBalance2 + morePayment;
+    auto gateway = root.create("gate", gatewayPayment);
+
+    // sets up gateway2 account
+    auto gateway2 = root.create("gate2", gatewayPayment);
+
+    auto idr = makeAsset(gateway, "IDR");
```
Comment (jonjove) [inline src/transactions/test/PathPaymentTests.cpp:4380]: Nitpick: `idr` and `usd` unused.

File: src/transactions/test/PathPaymentTests.cpp [hunk 4]
```diff
@@ -4442,3 +4339,147 @@ TEST_CASE("pathpayment", "[tx][pathpayment]")
...
+                                 .offers;
+                });
+            auto expected = std::vector<ClaimOfferAtom>{
+                o1.exchanged(640, 1280), o2.exchanged(320, 640),
+                o3.exchanged(160, 320),  o4.exchanged(80, 160),
+                o1.exchanged(40, 80),    o2.exchanged(20, 40),
+                o3.exchanged(10, 20)};
+            REQUIRE(actual == expected);
+            // clang-format off
+                market.requireBalances(
```
Comment (jonjove) [inline src/transactions/test/PathPaymentTests.cpp:4460]: Nitpick: gained an extra indent here

File: src/transactions/test/PathPaymentStrictSendTests.cpp (hunk 1 above)
Comment (rokopt) [reply src/transactions/test/PathPaymentStrictSendTests.cpp:2438]: Oops, thanks.  I was about to say that I had no idea how clang-format let me get away with that, but then I saw the "clang-format off".  :)  Fixed.

File: src/transactions/test/PathPaymentStrictSendTests.cpp (hunk 2 above)
Comment (rokopt) [reply src/transactions/test/PathPaymentStrictSendTests.cpp:2368]: Thanks -- now I'm going to say that I have no idea how clang let me get away with that!  :)

File: src/transactions/test/PathPaymentTests.cpp (hunk 3 above)
Comment (rokopt) [reply src/transactions/test/PathPaymentTests.cpp:4380]: Fixed.

File: src/transactions/test/PathPaymentTests.cpp (hunk 4 above)
Comment (rokopt) [reply src/transactions/test/PathPaymentTests.cpp:4460]: Fixed.

**Comment by rokopt** [comment]
I'm closing this PR because I've opened https://github.com/stellar/stellar-core/pull/2587 for a cleaner history with a single commit that integrates my fixes for Jon's review comments on this one.

**Pull request closed without merging**